src/core/service.c

```diff
@@ -2695,6 +2698,13 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
                                 /* Fall through */
 
                         case SERVICE_RUNNING:
+                                if (s->type == SERVICE_NOTIFY) {
```

Comment:
I cannot recall the reasoning behind putting it here, but I think I had one. Indeed I wanted to prevent `service_enter_running` from being called also when already running. I'll move the code up as I don't see any reason for this any more.